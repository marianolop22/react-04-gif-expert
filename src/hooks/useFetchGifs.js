import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {
    
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect( ()=> {

        getGifs( category )
            .then( images => {
                setState ({
                    data: images,
                    loading: false
                });
            })

    }, [category]); //al estar vac√≠o, no se vuelve a llamar la funcion, al poner algo, esto se ejecuta cada vez que se modifica



    return state;

};