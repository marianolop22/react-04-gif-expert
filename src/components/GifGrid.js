import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({category}) => {
/* 
    const [images, setImages] = useState([]);

    useEffect( ()=> {
        getGifs(category)
            .then( images => {
                setImages(images);
            })
    }, [category]); //al estar vac√≠o, no se vuelve a llamar la funcion, al poner algo, esto se ejecuta cada vez que se modifica
 */

    const { data:images, loading} = useFetchGifs( category );

    return (
        <>
            <h3 className="animate__animated animate__fadeIn" style={{margin: 10+'px'}}>{category}</h3>
            {loading && <p>Loading....</p>}
            <div className="card-grid">
                    {
                        images.map ( (img) => <GifGridItem 
                            key={img.id} 
                            {...img} 
                        /> )
                    }
            </div>
        </>
    )
}

export default GifGrid
