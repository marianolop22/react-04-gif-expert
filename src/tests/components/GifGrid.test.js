import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //voy a simular cualquier llamada a ese archivo

describe ('Pruebas del componente GifGrid', ()=> {

    const category = 'One Punch';
    
    test('debe renderizar el GifGrid', () => {

        //simulamos la llamada al usefetch y retorna estos valores
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const imgs = [
            {
                id: 'ABC',
                url: 'https://localhost/cualquiercosaaa/.jpg',
                title: 'un titulo'
            },
            {
                id: 'ABD',
                url: 'https://localhost/cualquiercosaaa/.jpg',
                title: 'un titulo'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect( wrapper.find('p').exists()).toBe(false); //aca verifico que no exista un parrafo de lo ading
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGridItem').length).toBe( imgs.length);
    });

});