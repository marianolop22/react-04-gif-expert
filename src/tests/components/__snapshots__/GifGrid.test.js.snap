// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGrid debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="un titulo"
      url="https://localhost/cualquiercosaaa/.jpg"
    />
    <GifGridItem
      id="ABD"
      key="ABD"
      title="un titulo"
      url="https://localhost/cualquiercosaaa/.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas del componente GifGrid debe renderizar el GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    One Punch
  </h3>
  <p>
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
